src/plugins/elasticsearch/index.js

```diff
@@ -1,38 +1,38 @@
-var _ = require('lodash');
-let Boom = require('boom');
+import { trim, trimRight } from 'lodash';
+import { methodNotAllowed } from 'boom';
 
-module.exports = function (kibana) {
-  var healthCheck = require('./lib/health_check');
-  var exposeClient = require('./lib/expose_client');
-  var createProxy = require('./lib/create_proxy');
+import healthCheck from './lib/health_check';
+import exposeClient from './lib/expose_client';
+import createProxy, { createPath } from './lib/create_proxy';
 
-  return new kibana.Plugin({
+module.exports = function ({ Plugin }) {
```

Comment:
I'd prefer to use `export default`, but it would technically be a BC break for this module.  It actually has a slightly different behavior when consuming the module via commonjs:

```js
// module.exports = function() {}
require('./path/to/elasticsearch') // function()

// export default function() {}
require('./path/to/elasticsearch') // { default: function() }
```

ES6 imports would do as you'd expect, but that would require changing all of the consuming files as well, which is beyond the scope here.